### PR TITLE
Improve error message for invalid quoted expression

### PR DESCRIPTION
**Pull request by mhanberg**
Addresses #8283.

Let me know what you think!

Thanks

### CODE REVIEW

File: lib/elixir/test/elixir/kernel/errors_test.exs [hunk 1]
```diff
@@ -706,7 +706,7 @@ defmodule Kernel.ErrorsTest do
 
   test "invalid macro" do
     assert_eval_raise CompileError,
-                      "nofile: invalid quoted expression: {:foo, :bar, :baz, :bat}",
+                      "nofile: invalid quoted expression: {:foo, :bar, :baz, :bat}\nYou may have meant to escape this with Macro.escape/1",
```
Comment (josevalim) [inline lib/elixir/test/elixir/kernel/errors_test.exs]: I would propose for us to use a regex so we don't have to change this every time. Something like this:

```suggestion
                      ~r"nofile: invalid quoted expression: {:foo, :bar, :baz, :bat}",
```

File: lib/elixir/src/elixir_expand.erl [hunk 2]
```diff
@@ -1116,7 +1116,10 @@ format_error({invalid_function_call, Expr}) ->
 format_error({invalid_call, Call}) ->
   io_lib:format("invalid call ~ts", ['Elixir.Macro':to_string(Call)]);
 format_error({invalid_quoted_expr, Expr}) ->
-  io_lib:format("invalid quoted expression: ~ts", ['Elixir.Kernel':inspect(Expr, [])]);
+  Message =
+    "invalid quoted expression: ~ts\n"
+    "You may have meant to escape this with Macro.escape/1",
```
Comment (josevalim) [inline lib/elixir/src/elixir_expand.erl]: I think we can be a bit more descriptive. Something like this:

```
"invalid quoted expression: %{foo: :bar}\n\n"
"Please make sure your quoted expressions are made of valid AST nodes."
"If you would like to introduce a value into the AST, such as a four-element"
tuple or a map, make sure to call `Macro.escape/1` before"
```

Thoughts?

Comment (mhanberg) [reply lib/elixir/src/elixir_expand.erl]: Sounds good to me! I'll update the branch.

**Comment by mhanberg** [comment]
@josevalim I have addressed your comments. Thanks for the feedback 👍

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**